xbmc/pvr/PVRManager.cpp

```diff
@@ -1579,9 +1571,8 @@ bool CPVRManager::PerformChannelSwitch(const CPVRChannelPtr &channel, bool bPrev
 
   CLog::Log(LOGDEBUG, "PVRManager - %s - switching to channel '%s'", __FUNCTION__, channel->ChannelName().c_str());
 
-  // will be deleted by CPVRChannelSwitchJob::DoWork()
```

Comment:
Hehe, good catch, <code>CPVRChannelSwitchJob::DoWork</code> actually did not (!) delete it, this was a bug, which now is fixed.